pkg/didcomm/dispatcher/outbound/outbound.go

```diff
@@ -158,16 +160,42 @@ func (o *Dispatcher) SendToDID(msg interface{}, myDID, theirDID string) error {
 			"outboundDispatcher.SendToDID failed to get didcomm destination for theirDID [%s]: %w", theirDID, err)
 	}
 
-	if len(connRec.ServiceEndPoint.Accept) > 0 {
-		dest.ServiceEndpoint.Accept = make([]string, len(connRec.ServiceEndPoint.Accept))
-		copy(dest.ServiceEndpoint.Accept, connRec.ServiceEndPoint.Accept)
-	}
-
 	mtp := o.mediaTypeProfile(dest)
 	switch mtp {
 	case transport.MediaTypeV1PlaintextPayload, transport.MediaTypeV1EncryptedEnvelope,
 		transport.MediaTypeRFC0019EncryptedEnvelope, transport.MediaTypeAIP2RFC0019Profile:
 		sendWithAnoncrypt = false
+
+		if len(connRec.MediaTypeProfiles) > 0 {
+			dest.MediaTypeProfiles = make([]string, len(connRec.MediaTypeProfiles))
+			copy(dest.MediaTypeProfiles, connRec.MediaTypeProfiles)
+		}
+	case transport.MediaTypeAIP2RFC0587Profile, transport.MediaTypeDIDCommV2Profile:
+		var (
+			url                 string
+			accept, routingKeys []string
+		)
+
+		url, err = connRec.ServiceEndPoint.URI()
```

Comment:
You trying to use ServiceEndPoint from connRec but the problem is the ServiceEndPoint fields are not exported so when we store in connRec in db as JSON the values of ServiceEndPoint are gone